broker.go

```diff
@@ -33,6 +34,10 @@ type Broker struct {
 	responses     chan *responsePromise
 	done          chan bool
 
+	// brokerAPIVersions stores the broker-advertised min/max API versions for each
+	// request key, if ApiVersionsRequest is enabled.
+	brokerAPIVersions map[int16][2]int16
```

Comment:
[premature optimization]🤔 This `[2]int16` does not save any space over a `struct{ min, max uin16 }` and the array is far less transparent about which is the minimum value, and which is the maximum value.

In fact, the `ApiVersionsResponseKey` itself is only 4 `int16` rather than 2. And there is not really going to be _that_ many entries anyways, right? So, without profiling indicating a need to optimize anything here, I would recommend that using a `map[int16]*ApiVersionsResponseKey` would be far more easy for humans to understand and keep straight.

Reply:
Updated